two_factor/views/core.py

```diff
@@ -199,6 +246,43 @@ def get_context_data(self, form, **kwargs):
             context['cancel_url'] = resolve_url(settings.LOGOUT_URL)
         return context
 
+    def get_remember_agent(self):
+        """
+        Returns if a user, browser and device is remembered using the remember cookie.
+        Uses a cached value if called before.
+        """
+        if self.remember_agent_cache is None:
+            self.remember_agent_cache = self._remember_agent()
+        return self.remember_agent_cache
```

Comment:
Django has a `cached_property` decorator which we should use rather than doing it ourselves: https://docs.djangoproject.com/en/3.0/ref/utils/#django.utils.functional.cached_property